src/mmw/js/src/modeling/models.js

```diff
@@ -53,48 +63,71 @@ var ProjectModel = Backbone.Model.extend({
-        // hold on to the scenarios so that we can
-        // save each one once we have an ID for the project
-        if (this.isNew) {
-            var tempScenarios = this.get('scenarios');
+    saveProjectAndScenarios: function() {
+        if (App.user.get('uid') !== this.get('uid')) {
+            return;
         }
 
-        this.save()
-            .done(function(resp) {
-                console.log('Saved project');
-
-                var scenarios = tempScenarios ? tempScenarios : self.get('scenarios');
-
-                scenarios.each(function(scenario) {
-                    if (!scenario.get('project')) {
+        var self = this,
+            scenarios = this.get('scenarios');
+        if (!this.get('id')) {
+            // We haven't saved the project before, save the project and then
+            // set the project ID on each scenario, then reattach the senarios
+            // to the project.
+            this.save()
+                .done(function() {
+                    scenarios.each(function(scenario) {
                         scenario.set('project', self.get('id'));
-                    }
-
-                    scenario
-                        .save()
-                        .done(function(resp) {
-                            console.log('Saved scenario');
-                        })
-                        .fail(function(resp) {
-                            console.log('Failed to save scenario');
-                        });
+                    });
+                })
+                .fail(function() {
+                    console.log('Failed to save project');
                 });
-            })
-            .fail(function(resp) {
-                console.log('Failed to save project');
+        } else {
+            this.save()
+                .fail(function() {
+                    console.log('Failed to save project');
+                });
+        }
+    },
+
+    addIdsToScenarios: function() {
+        if (!this.get('id')) {
+            this.saveProjectAndScenarios();
+        } else {
+            var self = this,
+                scenarios = this.get('scenarios');
+
+            scenarios.each(function(scenario) {
+                if (!scenario.get('project')) {
+                    scenario.set('project', self.get('id'));
+                }
             });
+        }
     },
 
     parse: function(response, options) {
-        var scenariosCollection = new ScenariosCollection();
+        if (response.scenarios) {
+            // If we returned scenarios (probably from a GET) then set them.
+            var uid = response.uid = response.user.id,
+                scenariosCollection = this.get('scenarios'),
+                scenarios = _.map(response.scenarios, function(scenario) {
+                    var modifications = _.map(scenario.modifications, function(modification) {
+                        return new ModificationModel(modification);
+                    });
```

Comment:
You may be able to just call `scenarioModel.get('modifications').reset(scenario.modifications);` (reset should cast raw objects to models).